### DISCUSSION TITLE
default_redirection_url in local compose deployment is incorrectly configured

### DISCUSSION THREAD

**Discussion by jonasrosland** [authelia/authelia#7153] [Documentation Feedback]
Having the following set in the `authelia/configuration.yml` file in the local compose example:
```
default_redirection_url: 'https://public.example.com'
```

results in authelia not starting, and erroring out with the following message:
```
authelia  | time="2024-04-13T04:31:27+10:00" level=error msg="Configuration: session: option 'cookies' must be configured with the per cookie option 'default_redirection_url' but the global one is configured which is not supported" stack="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_arm64.s:1222                                         goexit"
authelia  | time="2024-04-13T04:31:27+10:00" level=fatal msg="Can't continue due to the errors loading the configuration" stack="github.com/authelia/authelia/v4/internal/commands/context.go:327 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.0/command.go:968                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_arm64.s:1222                                         goexit"
```

Commenting the option out (I just wanted to see if that helps) makes the authelia local compose example work just fine.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The error you're encountering indicates that the `default_redirection_url` should not be configured globally under the `session` option in the `authelia/configuration.yml` file. Instead, it needs to be configured per cookie. This is a specific requirement of Authelia's configuration as it allows more granular control over the redirection behavior for different cookies.

To resolve this issue, you should remove the `default_redirection_url` from the global `session` settings and configure it within each specific cookie configuration under the `cookies` section. Here is an example of how you might configure it:

```yaml
session:
  name: 'authelia_session'
  secret: 'your_secret_here'
  expiration: 1h
  inactivity: 5m
  remember_me_duration: 1M
  domain: 'example.com'
  cookies:
    - domain: 'example.com'
      default_redirection_url: 'https://public.example.com'
```

Make sure to adjust the domain and other parameters according to your specific setup. This configuration ensures that the `default_redirection_url` is set specifically for the cookie related to the domain you specify, which aligns with Authelia's configuration requirements.

If you need further details on configuring sessions and cookies, you might want to refer to the official Authelia documentation or community discussions related to session configuration.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Thanks and fixed.

**Reply by jonasrosland** [reply]
Thank you!
